cpp/Platform.Delegates/Platform.Delegates.CreateDelegate.h

```diff
@@ -1,29 +1,27 @@
-#pragma once
-
-#ifndef PLATFORM_DELEGATES_CREATEDELEGATE
-#define PLATFORM_DELEGATES_CREATEDELEGATE
+#ifndef PLATFORM_DELEGATES_CREATEDELEGATE_H
+#define PLATFORM_DELEGATES_CREATEDELEGATE_H
 
 #include "Platform.Delegates.Delegate.h"
 
 namespace Platform::Delegates
 {
     template <typename ReturnType, typename... Args>
-    static Delegate<ReturnType(Args...)> CreateDelegate(ReturnType(&function)(Args...))
+    inline Delegate<ReturnType(Args...)> CreateDelegate(ReturnType(&function)(Args...))
     {
-        return Delegate<ReturnType(Args...)>(function);
+        return Delegate<ReturnType(Args...)>{function};
     }
 
     template <typename ReturnType, typename... Args>
-    static Delegate<ReturnType(Args...)> CreateDelegate(const std::function<ReturnType(Args...)> &function)
+    inline Delegate<ReturnType(Args...)> CreateDelegate(const std::function<ReturnType(Args...)> &function)
```

Comment:
Doesn`t C++ compiler inline methods by default?

Reply:
it doesn't. C++ compiler can [not] inline functions itself, and 'inline' specifier is just a recommendation. 'Inline' keyword is used here for "weak external" linkage, thanks to which multiple declaration does not violate One Definition Rule and it links only one function in all translation units.

Reply:
Do we really need explicit `inline` keyword here? Will the compiler do implicit inlining here without the keyword?

Reply:
https://en.wikipedia.org/wiki/One_Definition_Rule ... well, ok, lets try this.